app/domain/operations/products/import_service_area.rb

```diff
@@ -0,0 +1,151 @@
+        Success(sheet)
+      end
+
+      def import_records(sheet, year, row_data_begin, issuer_profile_hash, rating_area_model)
+        state_abbreviation = Settings.aca.state_abbreviation
+
+        if rating_area_model == 'single'
+          create_service_area_for_single_model(year, state_abbreviation)
+          return Success("Created Service Areas")
+        end
+
+        issuer_hios_id = sheet.cell(6,2).to_i.to_s
+
+        (row_data_begin..sheet.last_row).each do |i|
+          serves_entire_state = to_boolean(sheet.cell(i,3))
+          if serves_entire_state
+            service_area = ::BenefitMarkets::Locations::ServiceArea.where(active_year: year, issuer_provided_code: sheet.cell(i,1), county_zip_ids: [],
+                                                                          covered_states: [state_abbreviation], issuer_profile_id: issuer_profile_hash[issuer_hios_id], issuer_provided_title: sheet.cell(i,2)).first
+            if service_area.present?
+              service_area.issuer_hios_id = issuer_hios_id
+              service_area.save
+            else
+              ::BenefitMarkets::Locations::ServiceArea.create(active_year: year, issuer_provided_code: sheet.cell(i,1), covered_states: [state_abbreviation], county_zip_ids: [],
+                                                              issuer_hios_id: issuer_hios_id, issuer_profile_id: issuer_profile_hash[issuer_hios_id], issuer_provided_title: sheet.cell(i,2))
+            end
+          elsif serves_entire_state == false
+            existing_state_wide_areas = ::BenefitMarkets::Locations::ServiceArea.where(
+              active_year: year,
+              issuer_provided_code: sheet.cell(i,1),
+              issuer_hios_id: issuer_hios_id,
+              covered_states: [state_abbreviation]
+            )
+
+            if existing_state_wide_areas.blank?
+              county_name = extract_county_name(sheet.cell(i,4))
+              query_criteria = load_query(rating_area_model, sheet.cell(i,6), county_name, state_abbreviation)
+
+              records = ::BenefitMarkets::Locations::CountyZip.where(query_criteria)
+
+              service_area = ::BenefitMarkets::Locations::ServiceArea.where(
+                active_year: year,
+                issuer_provided_code: sheet.cell(i,1),
+                issuer_hios_id: issuer_hios_id,
+                issuer_profile_id: issuer_profile_hash[issuer_hios_id]
+              ).first
+
+              if service_area.blank?
+                ::BenefitMarkets::Locations::ServiceArea.create!({ active_year: year, issuer_provided_code: sheet.cell(i,1), issuer_hios_id: issuer_hios_id,
+                                                                   issuer_profile_id: issuer_profile_hash[issuer_hios_id], issuer_provided_title: sheet.cell(i,2), county_zip_ids: records.pluck(:_id)})
+              else
+                service_area.update_attributes(issuer_provided_title: sheet.cell(i,2), county_zip_ids: records.pluck(:_id))
```

Comment:
When updating with a new row on a service area that already exists - don't we need to include the county/zip pairs that were already present?  Wouldn't this code just wipe those out and only put the very last row in the spreadsheet on the record?